providers/xai/models/grok-4.toml

```diff
@@ -0,0 +1,23 @@
+name = "Grok 4"
+release_date = "2025-07-09"
+last_updated = "2025-07-09"
+attachment = false
+reasoning = true
+temperature = true
+knowledge = "2025-07"
+tool_call = true
+open_weights = false
+
+[cost]
+input = 3.00
+output = 15.00
+cache_read = 0.75
+cache_write = 15.00
+
+[limit]
+context = 128_000
+output = 16_384 # uncertain, based on grok 3
+
+[modalities]
+input = ["text"]
+output = ["text"]
```

Comment:
XAi presents conflicting info on this point

<img width="454" height="177" alt="Image" src="https://github.com/user-attachments/assets/41ea1a29-86c2-4fdf-b39b-fb51ce416407" />
<img width="708" height="310" alt="Image" src="https://github.com/user-attachments/assets/975602a7-c77d-475b-a4b3-cd2ddac75d26" />